app/models/agents/zendesk_satisfaction_ratings_agent.rb

```diff
@@ -0,0 +1,169 @@
+          * `retrieve_ticket` - If `true`, the agent wil use the `ticket_id`s received and find the associated `ticket`s
+          * `expected_receive_period_in_days` - Specify the period in days used to calculate if the agent is working.
+      MD
+    end
+
+    event_description <<-MD
+      Events look like this:
+      {
+        "id":              35436,
+        "url":             "https://company.zendesk.com/api/v2/satisfaction_ratings/35436.json",
+        "assignee_id":     135,
+        "group_id":        44,
+        "requester_id":    7881,
+        "ticket_id":       208,
+        "score":           "good",
+        "updated_at":      "2011-07-20T22:55:29Z",
+        "created_at":      "2011-07-20T22:55:29Z",
+        "comment":         "Awesome support!",
+        "user":            {...},
+        "ticket":          {...}
+      }
+    MD
+
+    form_configurable :subdomain
+    form_configurable :account_email
+    form_configurable :api_token
+    form_configurable :filter
+    form_configurable :mode, type: :array, values: %w(all on_change merge)
+    form_configurable :retrieve_assignee, type: :array, values: %w(true false)
+    form_configurable :retrieve_ticket, type: :array, values: %w(true false)
+    form_configurable :expected_update_period_in_days
+
+    def default_options
+      {
+        'subdomain' => 'myaccount',
+        'filter' => 'sort_order=desc&score=received_with_comment',
+        'account_email' => '{% credential ZendeskEmail %}',
+        'api_token' => '{% credential ZendeskToken %}',
+        'expected_update_period_in_days' => '2',
+        'mode' => 'on_change',
+        'retrieve_assignee' => 'true',
+        'retrieve_ticket' => 'true'
+      }
+    end
+
+    def validate_options
+      super
+
+      %w(subdomain account_email api_token).each do |key|
+        errors.add(:base, "The '#{key}' option is required.") if options[key].blank?
+      end
+
+      if boolify(options['retrieve_assignee']).nil?
+        errors.add(:base, "The retrieve_assignee option must be true or false")
+      end
+
+      if boolify(options['retrieve_ticket']).nil?
+        errors.add(:base, "The retrieve_ticket option must be true or false")
+      end
+    end
+
+    private
+
+    def build_default_options
+      options['url'] = "https://#{options['subdomain']}.#{DOMAIN}#{API_ENDPOINT}"
+      options['url'] << "?#{options['filter']}" if options['filter'].present?
+      options['basic_auth'] = "#{options['account_email']}/token:#{options['api_token']}"
+      options['type'] = 'json'
+      options['extract'] = EXTRACT
+
+      if boolify(options['retrieve_ticket'])
+        options['extract']['ticket'] = { 'path' => 'satisfaction_ratings.[*].ticket' }
+      end
+
+      if boolify(options['retrieve_assignee'])
+        options['extract']['user'] = { 'path' => 'satisfaction_ratings.[*].user' }
+      end
+    end
+
+    def parse(data)
```

Comment:
Hmm I didn't think on that :/

I think it will do it for the old ones too. That's not good.